### DISCUSSION TITLE
Reduce node-gyp cache size

### DISCUSSION THREAD

**Discussion by profbbrown** [eclipse-theia/theia#13340] [Q&A]
I've been running Thiea on top of a Docker container for a few years. I recently upgraded to the latest (1.46) from a much earlier version and one big difference is the node-gyp cache. The Docker container uses volumes to store the `/home` directory and the size of the volume has gone from a few MB to 500MB at minimum. This is putting a strain on my VM's disk usage, as I have about a hundred of these volumes to store.

Can the size of the cache be reduced? And if so, are there any downsides to it?

Alternatively, can I relocate the cache to, say, /tmp where it will be stored on the ephemeral filesystem?

**Comment by msujew** [comment]
Hey @profbbrown,

how come you store the node-gyp cache at all in your docker image? Theia doesn't require anything from the node-gyp cache during the runtime, especially if you enable backend bundling which has been added in a more recent version of Theia. The ideal way to use Theia in Docker would be to build your app in one layer and then move the build artifacts to a runtime layer without the `node_modules` or any of the other dev artifacts.

**Comment by profbbrown** [comment]
Well, the long answer is we developed our Docker images back in 2021 using the Theia version of that time. Because it had been working so well, we were reluctant to use more current versions. Now in 2024, we've rolled the version forward to 1.46 and have run into these issues. Since the .cache folder didn't exist (or wasn't nearly as big) in the older version, I assumed it was being created at runtime.

Is it OK to just delete the entire .cache folder after Theia has been built?

**Reply by msujew** [reply]
> Is it OK to just delete the entire .cache folder after Theia has been built?

I believe so. I would recommend to enable backend bundling in your app, see [here](https://github.com/eclipse-theia/theia/blob/master/examples/browser/webpack.config.js). That way you don't need any artifacts except for the app folder itself (mostly what's inside the `lib` folder).
